app/Services/Scheduler/EmailStatementService.php

```diff
@@ -92,6 +94,11 @@ private function calculateStartAndEndDates(): array
             EmailStatement::LAST_QUARTER => [now()->startOfDay()->subQuarterNoOverflow()->firstOfQuarter()->format('Y-m-d'), now()->startOfDay()->subQuarterNoOverflow()->lastOfQuarter()->format('Y-m-d')],
             EmailStatement::THIS_YEAR => [now()->startOfDay()->firstOfYear()->format('Y-m-d'), now()->startOfDay()->lastOfYear()->format('Y-m-d')],
             EmailStatement::LAST_YEAR => [now()->startOfDay()->subYearNoOverflow()->firstOfYear()->format('Y-m-d'), now()->startOfDay()->subYearNoOverflow()->lastOfYear()->format('Y-m-d')],
+            EmailStatement::ALL_TIME => [
+                $client->invoices()->selectRaw('MIN(invoices.date) as start_date')->pluck('start_date')->first()
```

Comment:
The whereIn on $client->invoices() can be brutal for performance. Can we replace with

```
Invoice::withTrashed()->where('client_id', $client->id)->selectRaw('MIN(invoices.date) as start_date')->pluck('start_date')->first()
```